sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelper.java

```diff
@@ -115,10 +115,12 @@
                 break;
 
             default:
-                String unknownAuthToken = "Unknown authorization token kind for read request";
+                String unknownAuthToken =
+                    "Unknown authorization token kind '" + originalRequestTokenType + "' for read request";
                 assert false : unknownAuthToken;
                 logger.error(unknownAuthToken);
-                throw Exceptions.propagate(new InternalServerErrorException(RMResources.InternalServerError));
+                throw Exceptions.propagate(
+                    new InternalServerErrorException(unknownAuthToken + " - " + RMResources.InternalServerError));
```

Comment:
This can be done in a follow up PR but do we have any UT validating that all case scenarios are handled? It's easy for a switch statement to be missing one.